### DISCUSSION TITLE
Change `escapeOnClose` input property after opening the dialog

### DISCUSSION THREAD

**Discussion by barremian** [primefaces/community#818] [PrimeNG] [answered]
Hi,
I'd like to change the `dialog` component's `escapeOnClose`  property after it has been opened. But it seems the the event listerner definition for `escapseOnClose` is bound to the [`@animation.start`](https://github.com/primefaces/primeng/blob/f955d12506bfdc06dfce5876bdcaa9a114f0165a/src/app/components/dialog/dialog.ts#L942) event. So it's triggered only at dialog open. My suggestion is to make the corresponding [`@Input()`](https://github.com/primefaces/primeng/blob/f955d12506bfdc06dfce5876bdcaa9a114f0165a/src/app/components/dialog/dialog.ts#L204) statement a setter and to trigger the event listener definition whenever the property is changed.

Would it be a feasible feature request?

I'm also willing to submit a PR if it's fine and obviously open for better suggestions/proposals.

Thanks

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

You can set `closeOnEscape` to `false`. There is no need to change the property after the dialog opens. Just disable it by default, output will be the same.

**Reply by barremian** [reply]
Hi @cetincakiroglu , thanks for the reply. This is our use-case:

1. Open dialog with a form -> here the dialog should be closeable on escape.
2. Fill the form and trigger HTTP form submit request.
3. During the HTTP request, we display an overlay throught the entire dialog like eg. "Processing..." with a spinner. When this overlay is shown, it should NOT be possible to close the dialog.
4. After the HTTP request is complete, close the dialog programatically and show a toast.

If you think that it seems unintuitive and not a practical use-case for dialog, do let me know. Thanks 🙂

**Reply by cetincakiroglu** [reply]
Could you please capture the escape event and stopPropagation or preventDefault? You basically need to cancel the event reaching the dialog.

**Reply by barremian** [reply]
Yes that could be a serviceable workaound. Thanks for the suggestion.
